apiserver/facades/client/application/deployrepository.go

```diff
@@ -362,11 +370,29 @@ func (v *deployFromRepositoryValidator) validate(arg params.DeployFromRepository
 	return dt, errs
 }
 
-func (v *deployFromRepositoryValidator) resolvedCharmValidation(resolvedCharm charm.Charm, arg params.DeployFromRepositoryArg) (deployTemplate, []error) {
+func validateAndParseAttachStorage(input []string, numUnits int) ([]names.StorageTag, []error) {
```

Comment:
I don't believe any test is hitting this code right? If so, we should probably add a test for this one.